atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionKeyValueServiceManager.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api;
+
+import com.palantir.atlasdb.cell.api.TransactionKeyValueService;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.processors.AutoDelegate;
+import java.util.function.LongSupplier;
+
+@AutoDelegate
+public interface TransactionKeyValueServiceManager extends AutoCloseable {
```

Comment:
I imagine this class will later on also host not just the TKVS, but also the DDL layer and the Sweep components for a given timestamp, right? (If not, you'll need to edit the constructors again later). If that's the case we might want to name this TimestampSwitchedKeyValueServiceComponents or something like that? (Or if not, we probably want to declare that and shift the thing being passed to the TM classes to have that?)

Reply:
hmmm, I was kinda hoping not to worry about the naming, as you said you are planning to rejig this. TransactionKeyValueServiceComponents sounds fine to me, but manager is kinda the most generic version or this?